File: src/053. Combinatoric selections/053.java
import java.math.BigInteger;
import java.util.ArrayList;

public class Main {
    public static void main(String[] args) {
        final int UPPER_LIMIT = 100;
        ArrayList<BigInteger> factorials = new ArrayList<>();
        factorials.add(BigInteger.ONE);
        for (int i = 1; i <= UPPER_LIMIT; i++) {
            factorials.add(factorials.get(i - 1).multiply(BigInteger.valueOf(i)));
        }

        int count = 0;
        for (int n = 1; n <= UPPER_LIMIT; n++) {
            for (int r = 1; r <= n; r++) {
                if (factorials.get(n).divide(factorials.get(r)).divide(factorials.get(n - r))
                        .compareTo(BigInteger.valueOf(1_000_000)) > 0) {
                    count += 1;
                }
            }
        }

        System.out.println(count);
    }
}

File: src/053. Combinatoric selections/053.go
package main

import (
	"fmt"
	"math/big"
)

func main() {
	const upperLimit = 100
	factorials := make([]*big.Int, upperLimit + 1)
	factorials[0] = big.NewInt(1)
	for i := 1; i <= upperLimit; i++ {
		var f big.Int
		f.Mul(factorials[i - 1], big.NewInt(int64(i)))
		factorials[i] = &f
	}

	count := 0
	for n := 1; n <= upperLimit; n++ {
		for r := 1; r <= n; r++ {
			var c big.Int
			c.Set(factorials[n])
			c.Div(&c, factorials[r])
			c.Div(&c, factorials[n - r])
			if c.Cmp(big.NewInt(1000000)) > 0 {
				count++
			}
		}
	}
	fmt.Println(count)
}